src/Components/Onboarding/Steps/Budget.tsx

```diff
@@ -1,70 +1,90 @@
 import * as React from "react"
+import { commitMutation, graphql } from "react-relay"
 import styled from "styled-components"
 
+import { ContextConsumer, ContextProps } from "../../Artsy"
 import SelectableToggle from "../SelectableToggle"
 import { StepProps } from "../Types"
 import { Layout } from "./Layout"
 
 const OptionsContainer = styled.div`
-width: 450px;
-margin: 0 auto 100px;
-&:last-child {
-  border-bottom: 1px solid #e5e5e5;
-}
+  width: 450px;
+  margin: 0 auto 100px;
+  &:last-child {
+    border-bottom: 1px solid #e5e5e5;
+  }
 `
 
 interface State {
-  selection: string
+  selection: number
 }
 
-export default class Budget extends React.Component<StepProps, State> {
-  options = [
-    "UNDER $500",
-    "UNDER $2,500",
-    "UNDER $5,000",
-    "UNDER $10,000",
-    "UNDER $25,000",
-    "UNDER $50,000",
-    "NO BUDGET IN MIND",
-  ]
+class Budget extends React.Component<StepProps & ContextProps, State> {
+  options = {
+    "UNDER $500": 500,
+    "UNDER $2,500": 2500,
+    "UNDER $5,000": 5000,
+    "UNDER $10,000": 10000,
+    "UNDER $25,000": 25000,
+    "UNDER $50,000": 50000,
+    "NO BUDGET IN MIND": 1000000000000,
+  }
 
   constructor(props) {
     super(props)
```

Comment:
Side-note, unfortunately when you override the `constructor`, you need to re-specify the typing of `props` otherwise those get lost completely.

```ts
type Props = StepProps & ContextProps

class Budget extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)
    this.state = {
      selection: null
    }
  }

  // ...
}
```

Or another thing you can do, if you do nothing in your override besides setting **static** initial state, is to use the syntax you used to define the `options` instance variable, thus not needing to override the constructor at all:

```ts
class Budget extends React.Component<Props, State> {
  state = {
    selection: null
  }

  // ...
}
```

Reply:
In this case would it be preferred to do both. Define the types and remove the constructor and define state as a static variable or is this an either or type of thing? I remember you pointing out the constructor thing out when pairing with Maxim and I awhile back. If I remember correctly removing the constructor is the preferred method. I'm just not 100% sure why.

Reply:
Yeah I would choose removing the constructor. Not because it’s inherently better than the other, jut because it leads to less code.